android/app/src/main/java/org/openbot/common/ControlsFragment.java

```diff
@@ -226,46 +231,38 @@ private void toggleIndicatorEvent(int value) {
     BotToControllerEventBus.emitEvent(ConnectionUtils.createStatus("INDICATOR_STOP", value == 0));
   }
 
-  @Override
-  public void onRequestPermissionsResult(
-      int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-    super.onRequestPermissionsResult(requestCode, permissions, grantResults);
-    switch (requestCode) {
-      case Constants.REQUEST_CONTROLLER_PERMISSIONS:
-        // If the permission is granted, start advertising to controller,
-        // otherwise, show a Toast
-        if (grantResults.length > 1
-            && (grantResults[0] == PackageManager.PERMISSION_GRANTED
-                && grantResults[1] == PackageManager.PERMISSION_GRANTED)) {
-          phoneController.connect(requireContext());
-        } else {
-          if (PermissionUtils.shouldShowRational(
-              requireActivity(), Constants.PERMISSION_LOCATION)) {
-            Toast.makeText(
-                    requireActivity().getApplicationContext(),
-                    R.string.location_permission_denied_controller,
-                    Toast.LENGTH_LONG)
-                .show();
-          }
-          if (PermissionUtils.shouldShowRational(
-              requireActivity(), Constants.PERMISSION_AUDIO_RECORDING)) {
-            Toast.makeText(
-                    requireActivity().getApplicationContext(),
-                    R.string.record_audio_permission_denied_controller,
-                    Toast.LENGTH_LONG)
-                .show();
-          }
-          if (PermissionUtils.shouldShowRational(requireActivity(), Constants.PERMISSION_CAMERA)) {
-            Toast.makeText(
-                    requireActivity().getApplicationContext(),
-                    R.string.camera_permission_denied_controller,
-                    Toast.LENGTH_LONG)
-                .show();
-          }
-          break;
-        }
-    }
-  }
+  protected final ActivityResultLauncher<String[]> requestPermissionLauncher =
+      registerForActivityResult(
+          new ActivityResultContracts.RequestMultiplePermissions(),
+          result -> {
+            result.forEach(
+                (permission, granted) -> {
+                  switch (permission) {
+                    case PERMISSION_CAMERA:
+                      if (!granted) {
+                        PermissionUtils.showCameraPermissionsControllerToast(requireActivity());
+                      }
+                      break;
+                    case PERMISSION_STORAGE:
```

Comment:
this is a master permission handler, to be accessed anywhere in app extending this class

Reply:
But I'm talking about the storage permission. The controller does not need storage permission afaik.

Reply:
i was using it as a global thing, so that if some other screen needs it, it could be used. but it's not used right now, so i've removed it.

Reply:
I understand if it's a general function, but here it is just for the ControllerFragment. It would be nice, if it was not required to have this function in each fragment.

Reply:
yeah so i've removed it